src/java.base/share/classes/java/math/MutableBigInteger.java

```diff
@@ -1892,6 +1913,204 @@ private boolean unsignedLongCompare(long one, long two) {
+                 * Thus, rad ≤ 2^ME is in the range of finite doubles.
+                 *
+                 * Noting that ex ≥ 0, we get bl-sh = P + ex ≥ P
+                 * which shows that x >> sh has at least P bits of precision,
+                 * since bl-sh is its bit length.
+                 */
+                // Shift the value into finite double range
+                rad = this.toBigInteger().shiftRight(rootSh * n).doubleValue();
+
+                // Use the root of the shifted value as an estimate.
+                // rad ≤ 2^ME, so Math.nextUp(rad) < Double.MAX_VALUE
+                rad = Math.nextUp(rad);
+                approx = nthRootApprox(rad, n);
+            } else { // fp arithmetic gives too few correct bits
+                // Set the root shift to the root's bit length minus 1
+                // The initial estimate will be 2^rootLen == 2 << (rootLen - 1)
+                rootSh = rootLen - 1;
+            }
+
+            if (rootSh == 0) {
+                // approx has at most ceil(Double.PRECISION / n) + 1 ≤ 19 integer bits
+                s = new MutableBigInteger((int) approx + 1);
+            } else {
+                // Allocate ⌈intLen / n⌉ ints to store the final root
+                s = new MutableBigInteger(new int[(intLen - 1) / n + 1]);
+
+                if (n >= Double.PRECISION) { // fp arithmetic gives too few correct bits
+                    // Set the initial estimate to 2 << (rootLen - 1)
+                    s.value[0] = 2;
+                    s.intLen = 1;
+                } else {
+                    // Discard wrong integer bits from the initial estimate
+                    // The radicand has Math.getExponent(rad)+1 integer bits, but only
+                    // the first Double.PRECISION leftmost bits are correct
+                    // We scale the corresponding wrong bits of approx in the fraction part.
+                    int wrongBits = ((Math.getExponent(rad) + 1) - Double.PRECISION) / n;
+                    // Since rad <= 2^(bitLength - sh), then
+                    // wrongBits <= ((bitLength - sh + 1) - Double.PRECISION) / n,
+                    // so wrongBits is less than ⌈(bitLength - sh) / n⌉,
+                    // the bit length of the exact shifted root,
+                    // hence wrongBits + rootSh < ⌈(bitLength - sh) / n⌉ + rootSh == rootLen
+                    rootSh += wrongBits;
+                    approx = Math.scalb(approx, -wrongBits);
+
+                    // now approx has at most ceil(Double.PRECISION / n) + 1 ≤ 19 integer bits
+                    s.value[0] = (int) approx + 1;
+                    s.intLen = 1;
+                }
+
+                /* The Newton's recurrence roughly duplicates the correct bits at each iteration.
+                 * Instead of shifting the approximate root into the original range right now,
+                 * we only duplicate its bit length and then refine it with Newton's recurrence,
```

Comment:
```suggestion
                /* The Newton's recurrence roughly doubles the correct bits at each iteration.
                 * Instead of shifting the approximate root into the original range right now,
                 * we only double its bit length and then refine it with Newton's recurrence,
```

Reply:
In `BigIntegerTest`, the methods related to nth-root do not seem to be invoked.

Reply:
If it makes your life easier, you could extract just your tests and create a new JUnit based test class.
Up to you.